OpenUtau/Views/MainWindow.axaml.cs

```diff
@@ -1057,6 +1061,40 @@ public partial class MainWindow : Window, ICmdSubscriber {
             DocManager.Inst.EndUndoGroup();
         }
 
+        void Transcribe(UPart part){
+            if(part is UWavePart wavePart){
+                try{
+                    using(Some some = new Some()){
+                        string text = ThemeManager.GetString("context.part.transcribing");
+                        var msgbox = MessageBox.ShowModal(this, $"{text} {part.name}", text);
+                        //Duration of the wave file in seconds
+                        int wavDurS = (int)(wavePart.fileDurationMs / 1000.0);
+                        var transcribeTask = some.Transcribe(DocManager.Inst.Project, wavePart, wavPosS =>{
+                            //msgbox?.SetText($"{text} {part.name}\n{wavPosS}/{wavDurS}");
+                            msgbox.SetText(string.Format("{0} {1}\n{2} / {3}",text, part.name, wavPosS, wavDurS));
+                        });
+                        transcribeTask.Wait();
```

Comment:
Instead of `Wait()` use `ContinueWith()`. See `RegenFrq()` for example. Modify project and show/hide dialog on scheduler from `TaskScheduler.FromCurrentSynchronizationContext()`.